### ISSUE TITLE
published=False returns published events

### ISSUE THREAD

**Issue by sbjzn** [MISP/PyMISP#1249]
PyMISP version: 2.4.194

The following code returns events that are published, even though published=False is specified.

```
misp = PyMISP(url=misp_url, key=misp_key, ssl=misp_ssl)
misp.global_pythonify = True

uuid = redacted-uuid
misp.search(
        controller="events",
        org=uuid,
        published=False,
        date_from="2024-01-01",
    )
```

**Comment by Rafiot** [comment]
Just making sure: you're getting both unpublished *and* published events, right?

This test is checking that, but there are no published events at all, so it's not gonna catch a bug where they're both returned: https://github.com/MISP/PyMISP/blob/main/tests/testlive_comprehensive.py#L864

**Comment by sbjzn** [comment]
Correct, I am getting both published and unpublished events returned.
